src/python/WMComponent/JobSubmitter/JobSubmitterPoller.py

```diff
@@ -709,9 +711,16 @@ def algorithm(self, parameters=None):
         2) Find jobs for all the necessary sites
         3) Submit the jobs to the plugin
         """
+        myThread = threading.currentThread()
+
+        if not self.passSubmitConditions:
+            msg = "JobSubmitter didn't pass the submit conditions. Skipping this cycle."
+            logging.warning(msg)
+            myThread.logdbClient.post("JobSubmitter_submitWork", msg, "warning")
```

Comment:
@ticoann Seangchan, it's probably much easier for you. All the unit tests failed with this error
`'_MainThread' object has no attribute 'logdbClient'`

I thought this attr would always be set for the components, since their super class is:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkerThreads/BaseWorkerThread.py#L150

Before I spend too much time on this, would you know how we can get this logdb defined for this thread?

Reply:
It is defined below,

https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkerThreads/BaseWorkerThread.py#L154

The problem was test configuration doesn't contain config.General.central_logdb_url
and that result in myThread.logdbClient  as None.
I think it is a little trick to fix the unittest. 
we have to create the logDB in unittest or mock it. (I prefer mocking it).
Anyway, it might be better to remove this log db update and I will create the separate issue for unittest to be passed.

Reply:
I might remove these logdb entries from the component then. I updated the test code to set central_logdb_url and it still fails to find that attr. Thanks, SEangchan!

Reply:
Thanks Alan, I will try to create the mock patch for this. I think we need those if we start to putting the log from different components